packages/super-components/components/msc-billing-tile/src/msc-billing-tile/msc-billing-commitment.tsx

```diff
@@ -0,0 +1,209 @@
+  OdsThemeTypographyLevel,
+  OdsThemeTypographySize,
+} from '@ovhcloud/ods-theming';
+import {
+  OdsIconName,
+  OdsIconSize,
+  OdsChipSize,
+  OdsHTMLAnchorElementTarget,
+  OdsChipVariant,
+} from '@ovhcloud/ods-core';
+import { translate, formatDate, Locale } from '@ovhcloud/msc-utils';
+import { ServiceDetails, Translations } from './msc-billing.types';
+import { getEngagementCommitUrl } from './urls';
+
+enum CommitmentStatus {
+  NONE = 'none',
+  ENDED = 'ended',
+  RENEWS = 'renews',
+  REQUESTED = 'requested',
+  ENDS = 'ends',
+}
+
+export interface IMscBillingCommitment {
+  serviceDetails?: ServiceDetails;
+  servicePath: string;
+  nextBillingDate?: string;
+  locale: Locale;
+  localeStrings: Translations;
+  commitmentDataTracking?: string;
+}
+
+@Component({
+  tag: 'msc-billing-commitment',
+  styleUrl: 'msc-billing-commitment.scss',
+  shadow: true,
+})
+export class MscBillingCommitment implements IMscBillingCommitment {
+  @Element() host!: HTMLStencilElement;
+
+  @Prop() servicePath: string;
+
+  @Prop() nextBillingDate: string;
+
+  @Prop() locale: Locale;
+
+  @Prop() localeStrings: Translations;
+
+  @Prop() serviceDetails?: ServiceDetails;
+
+  @Prop() commitmentDataTracking?: string;
+
+  @State() commitmentStatus: CommitmentStatus;
+
+  private getCommitmentStatus() {
+    if (
+      new Date(this.serviceDetails?.billing.engagement?.endDate || '') <
+      new Date()
+    ) {
+      return CommitmentStatus.ENDED;
+    }
+    if (
+      new Date(this.serviceDetails?.billing.engagement?.endDate || '') >
+      new Date()
+    ) {
+      return CommitmentStatus.RENEWS;
+    }
+    if (this.serviceDetails?.billing.engagementRequest) {
+      return CommitmentStatus.REQUESTED;
+    }
+    if (this.serviceDetails?.billing.expirationDate) {
+      return CommitmentStatus.ENDS;
+    }
+    return CommitmentStatus.NONE;
+  }
+
+  async componentWillLoad() {
+    this.commitmentStatus = this.getCommitmentStatus();
+  }
+
+  private getDescription(): string {
+    const labelByStatus = {
+      [CommitmentStatus.ENDED]: translate(
+        this.localeStrings,
+        'manager_billing_subscription_engagement_status_engaged_expired',
+        {
+          endDate: this.nextBillingDate,
+        },
+      ),
+      [CommitmentStatus.RENEWS]: translate(
+        this.localeStrings,
+        'manager_billing_subscription_engagement_status_engaged_renew',
+        {
+          endDate: this.nextBillingDate,
+        },
+      ),
+      [CommitmentStatus.REQUESTED]: translate(
+        this.localeStrings,
+        'manager_billing_subscription_engagement_status_commitement_pending',
+        {
+          nextBillingDate: formatDate(
+            this.serviceDetails?.billing.engagementRequest
+              ?.requestDate as string,
+            this.locale,
+          ),
+        },
+      ),
+      [CommitmentStatus.NONE]: '',
+      [CommitmentStatus.ENDS]: '',
+    };
+    return labelByStatus[this.commitmentStatus];
```

Comment:
**nitpick:** Since there can be only one status, no need to compute the description for all status.
Can be changed to `switch` statement.

Reply:
Yes good idea for better performance
Fixed on this branch : https://github.com/ovh/manager/pull/9986